backend/src/main/java/org/polsl/backend/service/HardwareService.java

```diff
@@ -87,4 +87,45 @@ public void createHardware(HardwareInputDTO request) {
     AffiliationHardware affiliationHardware = new AffiliationHardware(affiliation, hardware);
     affiliationHardwareRepository.save(affiliationHardware);
   }
+
+  @Transactional
+  public void editHardware(Long id, HardwareInputDTO request) throws NotFoundException {
+    Hardware hardware = hardwareRepository.findById(id)
+        .orElseThrow(() -> new NotFoundException("sprzęt", "id", id));
+    hardware.setName(request.getName());
+    HardwareDictionary hardwareDictionary = hardwareDictionaryRepository.findById(request.getDictionaryId())
+        .orElseThrow(() -> new NotFoundException("słownik urządzeń", "id", request.getDictionaryId()));
+    hardware.setHardwareDictionary(hardwareDictionary);
+    hardwareRepository.save(hardware);
+
+    if (request.getComputerSetId() != null) {
+      Set<ComputerSetHardware> computerSetHardwareSet = computerSetHardwareRepository.findAllByHardwareId(id);
+      if (computerSetHardwareSet != null && computerSetHardwareSet.size() > 0) {
+        ComputerSetHardware lastEntry = null;
+        for (Iterator setIterator = computerSetHardwareSet.iterator(); setIterator.hasNext(); ) {
+          lastEntry = (ComputerSetHardware) setIterator.next();
+        }
+        lastEntry.setValidTo(LocalDateTime.now());
+        computerSetHardwareRepository.save(lastEntry);
+      }
+
+      ComputerSet computerSet = computerSetRepository.findById(request.getComputerSetId())
+          .orElseThrow(() -> new NotFoundException("zestaw komputerowy", "id", request.getComputerSetId()));
+      ComputerSetHardware computerSetHardware = new ComputerSetHardware(computerSet, hardware);
+      computerSetHardwareRepository.save(computerSetHardware);
+    }
+
+    Set<AffiliationHardware> affiliationHardwareSet = affiliationHardwareRepository.findAllByHardwareId(id);
+    AffiliationHardware lastEntryAffiliation = null;
+    for (Iterator setIterator = affiliationHardwareSet.iterator(); setIterator.hasNext(); ) {
+      lastEntryAffiliation = (AffiliationHardware) setIterator.next();
+    }
+    lastEntryAffiliation.setValidTo(LocalDateTime.now());
```

Comment:
Jeszcze tylko jedna pierdoła - środowisko informuje, że tu może powstać `NullPointerException`. Nie wiem, czy jest to poważny problem, ale jeśli się da, to uniknijmy tego 😉